include/Dialect/Secret/IR/SecretOps.h

```diff
@@ -16,4 +16,24 @@
 #define GET_OP_CLASSES
 #include "include/Dialect/Secret/IR/SecretOps.h.inc"
 
+namespace mlir {
+namespace heir {
+namespace secret {
+
+// Extracts the given op from inside the generic body and lifting to a new
+// single-op generic after the context generic op.
+//
+// Replaces `genericOp` with a new genericOp using `rewriter`, and returns
+// the two newly created generic ops, with the first one being the replacement
+// for the input `genericOp`, and the second one being the extracted genericOp.
+//
+// Handles adding the operands of opToExtract to the yielded values of the
+// generic.
+std::pair<GenericOp, GenericOp> extractOpAfterGeneric(
```

Comment:
nit: could you add a quick comment that this must be used only used when the `opToExtract`'s results do not have any further uses besides in the yield of the `genericOp`?

Reply:
Done